bmaptools/TransRead.py

```diff
@@ -206,16 +206,17 @@ def _read_thread(self, f_from, f_to):
         """
 
         chunk_size = 1024 * 1024
-        while not self._done:
-            buf = f_from.read(chunk_size)
-            if not buf:
-                break
-
-            f_to.write(buf)
-
-        # This will make sure the process decompressor gets EOF and exits, as
-        # well as ublocks processes waiting on decompressor's stdin.
-        f_to.close()
+        try:
```

Comment:
Perhaps cleaner to use the fact that file-like objects are context managers:

```
with f_to:
    while not self._done:
        ...
```

or if `f_to` is not guaranteed to be a context manager:

```
from contextlib import closing

with closing(f_to):
    while not self._done:
        ...
```

Reply:
I agree. I was very new to python when I was writing this code, and if I had to do it today, it would look differently, and I'd use 'with' and context manager extensively.

Let's see if the author can incorporate your suggestion.